### PR TITLE
Add evaluation test to regression testing

### PR DESCRIPTION
**Pull request by asyatrhl**

### CODE REVIEW

**Review by MaximGorkem — COMMENTED** [review]
Looks good.

File: regression/create_test_script.py [hunk 1]
```diff
@@ -83,18 +83,26 @@ def joining(lst):
...
 
-                data_name = temp[k+1]
-                if data_name in config and "datapath" in config[data_name]:
+                try:
                     path_data = config[log_data]["datapath"]
-                    temp.insert(-1, '--data ' + path_data)
+                    temp[i+1] = str(config[log_data][log_model]["epoch"])
+                except KeyError:
+                    # Handle the KeyError by assigning a dummy value
+                    path_data = "/data_ssd"
```
Comment (ermanok) [inline regression/create_test_script.py:102]: it seems like the default path is the oath in the server. isn't it better for the script to give an exception and exit when the data path would not be given in the config file?

File: regression/last_dev.py [hunk 2]
```diff
@@ -91,18 +91,25 @@ def dev_scripts(script_pth, output_file_pth):
...
 
-                    data_name = temp[k+1]
-                    if data_name in config and "datapath" in config[data_name]:
+                    try:
                         path_data = config[log_data]["datapath"]
-                        temp.insert(-1, '--data ' + path_data)
+                        temp[i+1] = str(config[log_data][log_model]["epoch"])
+                    except KeyError:
+                        # Handle the KeyError by assigning a dummy value
+                        path_data = "/data_ssd"
```
Comment (ermanok) [inline regression/last_dev.py:110]: same comment with the previous one about the exception

**Review by ermanok — CHANGES_REQUESTED** [review]

**Review by rotx-maxim — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -5,8 +5,48 @@ on:
...
 
 jobs:
+  eval:
+    runs-on: self-hosted
+    timeout-minutes: 345600
+    steps:
+      - name: Checkout last-dev
+        uses: actions/checkout@v2
+        with:
+          repository: asyatrhl/ai8x-training
```
[the remark this answers is not in the document]
Comment (asyatrhl) [reply .github/workflows/test.yml]: Fixed.

File: regression/last_dev.py (hunk 2 above)
Comment (asyatrhl) [reply regression/last_dev.py]: Handled with a warnings message.

File: regression/create_test_script.py (hunk 1 above)
Comment (asyatrhl) [reply regression/create_test_script.py]: Handled with a warnings message.

**Review by ermanok — APPROVED** [review]
That's good.

**Review by rotx-maxim — APPROVED** [review]

**Pull request merged**